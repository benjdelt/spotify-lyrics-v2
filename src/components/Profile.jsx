import { useTranslation } from 'react-i18next';
import { Dropdown, DropdownMenuItem } from './Dropdown';
import Loader from './Loader';
import '../styles/Profile.css';

function Profile({user, loading, logout}) {

  const { t } = useTranslation();

  function UserLabel() {
    return (
      <>
        <img src={user.avatarUrl} alt="avatar" className="avatar"/>
        <span className="user text">
          {user.name}
        </span>
      </>
    );
  }

  return (
    <section className="profile">
      {loading ? (
        <Loader />  
      ) : (
        user.name ? (
          <Dropdown label={<UserLabel />}>
              <DropdownMenuItem key={1} handleClick={logout} >{t('user.logout')}</DropdownMenuItem>
          </Dropdown>
        ) : (
          <a href="/.netlify/functions/login" className="login">{ t('login.button') }</a>
      ))}
    </section>
  );
}

export default Profile;
