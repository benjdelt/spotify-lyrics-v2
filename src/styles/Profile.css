.profile {
  display: flex;
  position: fixed;
  top: 1em;
  right: 1em;
}

.login {
  color: #fff;
  justify-content: center;
  padding: 11px 44px;
  border: 2px solid #fff;
  border-radius: 500px;
  font-weight: 600;
  text-decoration: none;
  align-self: center;
  -webkit-font-feature-settings: "c2sc";
  font-feature-settings: "c2sc";
  font-variant: small-caps;
}

.login:hover, .login:focus {
  -webkit-transform: scale(1.06);
  transform: scale(1.06);
  outline: none;
}

.user {
  display: flex;
  align-items: center;
  font-weight: bold;
}

.avatar {
  width: 2em;
  border-radius: 50%;
  margin-right: .5em;
}

