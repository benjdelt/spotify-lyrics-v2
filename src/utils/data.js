import SpotifyWebApi from 'spotify-web-api-js';

const spotifyApi = new SpotifyWebApi();


const getHashParams = (location = window.location) => {
  const q = location.hash.substring(2);

  // if the substring is non-empty, split it up based on hashes
  // return the parameters as an object
  if (q) {
    const qArray = q.split("/");
    const hashParams = {
      path: qArray[0],
      access_token: qArray[1],
      refresh_token: qArray[2],
      expires_in: qArray[3],
    };
    return hashParams;
  }
  return {};
};

const params = getHashParams();
const token = params.access_token;
if (token) {
  spotifyApi.setAccessToken(token);
}

export const getUser = async () => {
  const response = await spotifyApi.getMe();
  return {name: response.display_name, avatar: response.images[0].url};
}
