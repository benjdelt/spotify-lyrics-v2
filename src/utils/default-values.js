const errorTypes = {
  noLyricsFound: "noLyricsFound",
  noCurrentlyPlaying: "noCurrentlyPlaying",
}

const emptyUser = {
  name: "",
  avatarUrl: "",
};

const emptySong = {
  title: "",
  artist: "",
  album: "",
  coverUrl: "",
  lyrics: [],
};

const defaultLoading = {
  user: true,
  song: true,
};

export {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading,
}
