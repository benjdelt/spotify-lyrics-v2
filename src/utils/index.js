import {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading
} from "./default-values";

export {
  errorTypes,
  emptyUser,
  emptySong,
  defaultLoading,
}
