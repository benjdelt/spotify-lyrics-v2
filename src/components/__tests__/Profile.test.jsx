import { render, cleanup } from '@testing-library/react';
import Profile from '../Profile';
import { emptyUser } from '../../utils/';

afterEach(cleanup)

it('should take a snapshot', () => {
  const { asFragment } = render(<Profile user={emptyUser} />)
  expect(asFragment(<Profile user={emptyUser} />)).toMatchSnapshot()
});
