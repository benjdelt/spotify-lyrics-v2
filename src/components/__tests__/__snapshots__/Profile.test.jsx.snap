// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="profile"
  >
    <a
      class="login"
      href="/.netlify/functions/login"
    >
      login.button
    </a>
  </section>
</DocumentFragment>
`;
